instance/templates/instance/memcached/memcached.nomad

```diff
@@ -0,0 +1,53 @@
+# Jobs need a globally consistent name, so we need to include the instance id.
+job "memcached-1234" {
+  datacenters = ["dc1"]
+  type = "service"
+  constraint {
+    distinct_hosts = true
+  }
+  update {
+    max_parallel = 1
+  }
+  migrate {
+    max_parallel = 1
+  }
+  group "memcached" {
+    count = 3    # Only 1 for stage
```

Comment:
Does this mean 3 memcached processes per instance? Wouldn't it be enough to go with 1, and have Nomad reschedule the job in case the node dies? (I'm assuming that's how things work).

Reply:
Oh wait, I realized that may not be practical because it'd mean reprovisioning the appserver with new settings since the ports will likely change.

Reply:
The goal is high availability.  With just a single process, the instance would be down while Nomad reschedules the process when it goes down, so we need at least two.  To still retain this resilience even during a rolling update or other maintenance work, three processes are what is generally recommended.  I'm fine with using either two or three processe, but I'd start with three.  If resource usage becomes an issue, we can reduce this to two.

The app server needs to be able to deal with changing ports.  The solution suggested in the original design is to use three fixed local ports on each app server that are dynamically forwarded to the correct ports via an iptables hack, with Consul watches in place to update the port forwards as needed.  However, now that Consul has the nice connect feature, we may be able to use that feature instead – I haven't looked into the details yet, but since this is exactly what it is meant to be used for, I'm optimistic that it will provide a nicer solution for the problem, with the added benefit of using TLS.

Reply:
OK, yeah I get the HA part now, not sure why I thought 1 was going to be okay :)

And yeah the proxy-that-we-manage solution seems a little complex, so I also hope Connect works just fine here.